### PR TITLE
created GetUserApps to return apps based on owner OR Contributors

### PR DESCRIPTION
**Pull request by satti-hari-krishna-reddy**

### CODE REVIEW

**Comment by satti-hari-krishna-reddy** [comment]
dont merge it

**Comment by satti-hari-krishna-reddy** [comment]
@frikky I successfully tested this process on our on-premises environment. However , i dont know if this works the same for cloud : )

File: db-connector.go [hunk 1]
```diff
@@ -5627,6 +5627,140 @@ func fixAppAppend(allApps []WorkflowApp, innerApp WorkflowApp) ([]WorkflowApp, W
...
+			return []WorkflowApp{}, err
+		}
+
+		for _, hit := range wrapped.Hits.Hits {
+			innerApp := hit.Source
+			userApps = append(userApps, innerApp)
+		}
+	} else {
+		cursorStr := ""
+		query := datastore.NewQuery(indexName).Filter("owner =", userId).Filter("contributors IN", []string{userId}).Order("-edited")
```
Comment (frikky) [inline db-connector.go:5718]: Test whether the query actually returns something properly on cloud as well

File: db-connector.go [hunk 2]
```diff
@@ -5627,6 +5627,140 @@ func fixAppAppend(allApps []WorkflowApp, innerApp WorkflowApp) ([]WorkflowApp, W
...
+				"bool": map[string]interface{}{
+					"should": []map[string]interface{}{
+						{
+							"match": map[string]interface{}{
+								"owner": userId,
+							},
+						},
+						{
+							"terms": map[string]interface{}{
+								"contributors": []string{userId},
```
Comment (frikky) [inline db-connector.go:5667]: Did you test that this actually works, and not just the "owner" part?

**Comment by frikky** [comment]
Looks good! Lets see how it fares in test.

**Pull request merged**